packages/@aws-cdk/aws-vpcv2-alpha/lib/route.ts

```diff
@@ -0,0 +1,788 @@
+    }
+
+    this.routerId = this.resource.attrVpnGatewayId;
+    this.vpcId = props.vpcId;
+  }
+}
+
+/**
+ * Creates a network address translation (NAT) gateway
+ * @resource AWS::EC2::NatGateway
+ */
+export class NatGateway extends Resource implements IRouteTarget {
+  /**
+   * The type of router used in the route.
+   */
+  readonly routerType: RouterType;
+
+  /**
+   * The ID of the route target.
+   */
+  readonly routerId: string;
+
+  /**
+   * [Public NAT gateway only] The allocation ID of the Elastic IP address that's
+   * associated with the NAT gateway. This property is required for a public NAT
+   * gateway and cannot be specified with a private NAT gateway.
+   * @default none
+   */
+  public readonly allocationId?: string;
+
+  /**
+   * Indicates whether the NAT gateway supports public or private connectivity.
+   * @default public
+   */
+  public readonly connectivityType?: string;
+
+  /**
+   * The maximum amount of time to wait before forcibly releasing the
+   * IP addresses if connections are still in progress.
+   * @default 350 seconds
+   */
+  public readonly maxDrainDuration?: Duration;
+
+  /**
+   * The NAT gateway CFN resource.
+   */
+  public readonly resource: CfnNatGateway;
+
+  constructor(scope: Construct, id: string, props: NatGatewayProps) {
+    super(scope, id);
+
+    this.routerType = RouterType.NAT_GATEWAY;
+
+    this.connectivityType = props.connectivityType || 'public';
+    this.maxDrainDuration = props.maxDrainDuration || Duration.seconds(350);
+
+    // If user does not provide EIP, generate one for them
+    var aId: string | undefined;
+    if (this.connectivityType == 'public') {
+      if (!props.allocationId) {
+        let eip = new CfnEIP(this, 'EIP', {
+          domain: props.vpcId,
+        });
+        aId = eip.attrAllocationId;
+      } else {
+        aId = props.allocationId;
+      }
+    }
+
+    this.resource = new CfnNatGateway(this, 'NATGateway', {
+      subnetId: props.subnet.subnetId,
+      allocationId: aId,
+      maxDrainDurationSeconds: props.maxDrainDuration?.toSeconds(),
+      ...props,
+    });
+
+    this.routerId = this.resource.attrNatGatewayId;
+
+    this.node.addDependency(props.subnet.internetConnectivityEstablished);
+  }
+}
+
+/**
+ * Creates a network interface
+ * @resource AWS::EC2::NetworkInterface
+ */
+export class NetworkInterface extends Resource implements IRouteTarget {
+  /**
+   * The type of router used in the route.
+   */
+  readonly routerType: RouterType;
+
+  /**
+   * The ID of the route target.
+   */
+  readonly routerId: string;
+
+  /**
+   * The network interface CFN resource.
+   */
+  public readonly resource: CfnNetworkInterface;
+
+  constructor(scope: Construct, id: string, props: NetworkInterfaceProps) {
+    super(scope, id);
+
+    this.routerType = RouterType.NETWORK_INTERFACE;
+
+    this.resource = new CfnNetworkInterface(this, 'NetworkInterface', {
+      subnetId: props.subnet.subnetId,
+    });
```

Comment:
there are too many properties defined in the [`AWS::EC2::NetworkInterface` resource](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-networkinterface.html), are we going to not support them

Reply:
note: remove this and do proper L2 implementation to support this